convert2rhel/actions/__init__.py

```diff
@@ -230,63 +285,199 @@ def get_actions(actions_path, prefix):
+        # its deps.  We don't care about the return value here.
+        actions_so_far = list(resolve_action_order(self.actions))
+
+        if self.next_stage:
+            self.next_stage.check_dependencies(actions_so_far)
+
+    def run(self, successes=None, failures=None, skips=None):
+        """
+        Run all the actions in Stage and other linked Stages.
+
+        :keyword successes: Actions which have already run and succeeded.
+        :type: Sequence
+        :keyword failures: Actions which have already run and failed.
+        :type: Sequence
+        :return: 2-tuple consisting of two lists.  One with Actions that
+            have succeeded and one of Actions that have failed.  These
+            lists contain the Actions passed in via successes and failures
+            in addition to any that were run in this :class`Stage`.
+
+        :rtype: FinishedActions
+        .. note:: Success is currently defined as an action whose status after
+            running is WARNING or better (WARNING or SUCCESS) and
+            failure as worse than WARNING (OVERRIDABLE, ERROR, FATAL)
+        """
+        logger.task("Prepare: %s" % self.task_header)
+
+        if self._has_run:
+            raise ActionError("Stage %s has already run." % self.stage_name)
+
+        # Default values and make copies so we don't overwrite callers' data
+        if successes is None:
+            successes = []
+        else:
+            successes = list(successes)
+
+        if failures is None:
+            failures = []
+        else:
+            failures = list(failures)
+
+        if skips is None:
+            skips = []
+        else:
+            skips = list(skips)
+
+        for action_class in resolve_action_order(self.actions):
+            # Decide if we need to skip because deps have failed
+            failed_deps = [d for d in action_class.dependencies if d not in successes]
+
+            action = action_class()
+
+            if failed_deps:
+                to_be = "was"
+                if len(failed_deps) > 1:
+                    to_be = "were"
+                message = "Skipped because %s %s not successful" % (format_sequence_as_message(failed_deps), to_be)
+
+                action.set_result(status="SKIP", error_id="SKIP", message=message)
+                skips.append(action)
+                continue
+
+            # Run the Action
+            try:
+                action.run()
+            except (Exception, SystemExit) as e:
+                # Uncaught exceptions are handled by constructing a generic
+                # failure message here that should be reported
+                message = (
+                    "Unhandled exception was caught: %s"
+                    "\nPlease file a bug at https://issues.redhat.com/ to have this"
+                    " fixed or a specific error message added."
+                    "\nTraceback: %s" % (e, traceback.format_exc())
```

Comment:
Not a necessary to apply, but I would format the message in this way, as it seems easier to read from the point-of-view that we start reading from left -> right, moving the `\n` and `" "` to the end of the string instead of the beginning. 

But again, not necessary to apply if you don't think it makes sense.

```suggestion
                    "Unhandled exception was caught: %s\n"
                    "Please file a bug at https://issues.redhat.com/ to have this "
                    "fixed or a specific error message added.\n"
                    "Traceback: %s" % (e, traceback.format_exc())
```